tools/regression_test.py

```diff
@@ -0,0 +1,980 @@
+        if (len(using_dataset) > 1) and (dataset not in using_dataset):
+            continue
+
+        dataset_type += f'{dataset} | '
+        task_name += f'{metric.get("Task", "")} | '
+
+        # remove some metric which not in metric_info from test yaml
+        for k, v in pytorch_meta_metric.items():
+
+            if k not in metric_name_info and 'Restorers' not in task_name:
+                continue
+
+            if 'Restorers' in task_name and k not in dataset_type:
+                # mmedit
+                continue
+
+            if isinstance(v, dict):
+                # mmedit
+                for sub_k, sub_v in v.items():
+                    use_metric = {sub_k: sub_v}
+                    metric_list.append(use_metric)
+                    pytorch_metric.update(use_metric)
+            else:
+                use_metric = {k: v}
+                metric_list.append(use_metric)
+                pytorch_metric.update(use_metric)
+
+    dataset_type = dataset_type[:-3].upper()  # remove the final ' | '
+    task_name = task_name[:-3]  # remove the final ' | '
+
+    # update useless metric
+    metric_all_list = [str(metric) for metric in metric_name_info]
+    metric_useless = set(metric_all_list) - set(
+        [str(metric) for metric in pytorch_metric])
+    for metric in metric_useless:
+        metric_list.append({metric: '-'})
+
+    # get pytorch fps value
+    fps_info = model_info.get('Metadata').get('inference time (ms/im)')
+    if fps_info is None:
+        fps = '-'
+    elif isinstance(fps_info, list):
+        fps = fps_info[0].get('value')
+    else:
+        fps = fps_info.get('value')
+
+    logger.info(f'Got metric_list = {metric_list} ')
+    logger.info(f'Got pytorch_metric = {pytorch_metric} ')
+
+    # update report
+    update_report(
+        report_dict=report_dict,
+        model_name=model_name,
+        model_config=str(model_config_name),
+        task_name=task_name,
+        model_checkpoint_name=str(checkpoint_path),
+        dataset=dataset_type,
+        backend_name='Pytorch',
+        deploy_config='-',
+        static_or_dynamic='-',
+        precision_type='-',
+        conversion_result='-',
+        fps=fps,
+        metric_info=metric_list,
+        test_pass='-',
+        report_txt_path=report_txt_path)
+
+    logger.info(f'Got {model_config_name} metric: {pytorch_metric}')
+    return pytorch_metric
+
+
+def get_info_from_log_file(info_type, log_path, yaml_metric_key=None):
+    """Get fps and metric result from log file.
+
+    Args:
+        info_type (str): Get which type of info: 'FPS' or 'metric'
+        log_path (str): Logger path.
+        yaml_metric_key (str): Name of metric from yaml metric_key.
+
+    Returns:
+        Float: Info value which get from logger file
+    """
+    log_path = Path(log_path)
+
+    if log_path.exists():
+        with open(log_path, 'r') as f_log:
+            lines = f_log.readlines()
+    else:
+        print(f'{log_path} do not exist !!!')
```

Comment:
Can we use logger here?

Reply:
Done. I replace all `print` to `logger.xxx`